src/api/services/iuf/sessions.go

```diff
@@ -508,13 +522,29 @@ func (s iufService) getDagTasks(session iuf.Session, stageInfo iuf.Stage) ([]v1a
 	if stageInfo.Type == "product" {
 		for _, product := range session.Products {
 			for index, operation := range stageInfo.Operations {
+				if !templateMap[operation.Name] {
+					s.logger.Warnf("The template %v cannot be found in Argo. Make sure you have run upload-rebuild-templates.sh from docs-csm", operation.Name)
+					continue
+				}
+
 				task := v1alpha1.DAGTask{
 					Name: product.Name + "-" + operation.Name,
 				}
 				// dep with a stage
 				if index != 0 {
-					task.Dependencies = []string{
-						product.Name + "-" + stageInfo.Operations[index-1].Name,
+					// check the last dependency is available
```

Comment:
A better comment here might be:

```suggestion
					// Find the previous operation for which a workflow template exists
```

Reply:
Done thanks.